catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/GeoJSONFeatureExtractor.java

```diff
@@ -0,0 +1,105 @@
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.geocoding.feature;
+
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.simplify.DouglasPeuckerSimplifier;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.spatial.geocoding.FeatureExtractionException;
+import org.codice.ddf.spatial.geocoding.FeatureExtractor;
+import org.codice.ddf.spatial.geocoding.FeatureIndexingException;
+import org.geotools.feature.FeatureIterator;
+import org.geotools.geojson.feature.FeatureJSON;
+import org.opengis.feature.simple.SimpleFeature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GeoJSONFeatureExtractor implements FeatureExtractor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GeoJSONFeatureExtractor.class);
+
+  private static final int MAX_SIMPLIFY_ITERATIONS = 5;
+
+  private static final double SIMPLIFY_DISTANCE_TOLERANCE = .1;
+
+  private static final int MAX_POINTS_PER_FEATURE = 500;
+
+  @Override
+  public void pushFeaturesToExtractionCallback(
+      String resource, ExtractionCallback extractionCallback) throws FeatureExtractionException {
+    Validate.notNull(extractionCallback, "extractionCallback can't be null");
+    Validate.notNull(resource, "resource can't be null");
+
+    FeatureIterator<SimpleFeature> iterator = getFeatureIterator(resource);
+
+    try {
+      while (iterator.hasNext()) {
+        SimpleFeature feature = iterator.next();
+        Geometry geometry = (Geometry) feature.getDefaultGeometry();
+        if (geometry == null) {
+          LOGGER.debug("Failed to get geometry from {}", resource);
+          continue;
+        }
+
+        Geometry simplifiedGeometry = getSimplifiedGeometry(geometry);
+
+        if (simplifiedGeometry == null) {
+          LOGGER.debug(
+              "Failed to simplify geometry below {} point maximum", MAX_POINTS_PER_FEATURE);
+        } else {
+          feature.setDefaultGeometry(simplifiedGeometry);
+          extractionCallback.extracted(feature);
+        }
+      }
+    } catch (FeatureIndexingException e) {
+      throw new FeatureExtractionException("Unable to extract feature from " + resource, e);
+    } finally {
+      iterator.close();
+    }
+  }
+
+  private FeatureIterator<SimpleFeature> getFeatureIterator(String resource)
+      throws FeatureExtractionException {
+    try {
+      File file = new File(resource);
+      if (file.length() == 0) {
+        throw new FeatureExtractionException("Empty resource");
+      }
+
+      Reader reader = new InputStreamReader(new FileInputStream(file), "UTF-8");
+      return new FeatureJSON().streamFeatureCollection(reader);
```

Comment:
It looks like the input stream doesn't get closed. There's nothing in the docs for streamFeatureCollection indicating that the reader is automatically closed. Probably need to put the reader in a try-with-resource block.

Reply:
Updated to close the stream